src/utils.c

```diff
@@ -1798,7 +1798,13 @@ bool
 as_key_move(as_key* dst, as_key* src)
 {
 	*dst = *src;
-	if (as_load_uint32(&src->value.integer._.count) > 1) {
+	if (!src->valuep) {
+		return true;
+	}
+	if (as_load_uint32(&src->valuep->integer._.count) > 1 || \
+		as_load_uint32(&src->valuep->string._.count) > 1 || \
+		as_load_uint32(&src->valuep->bytes._.count) > 1 ) {
```

Comment:
Per Brian's information lets change this back to a single check on integer._.count.